File: benchmarks/sum_of_squares/sum_of_squares.c
#include <stdio.h>

long long square(long long n) { return n * n; }

int main(void) {
  long long total = 0;
  long long i = 0;
  while (i < 1000000LL) {
    total += square(i);
    i += 1;
  }
  printf("%lld\n", total);
  return 0;
}

File: benchmarks/sum_of_squares/sum_of_squares.rs
fn square(n: i64) -> i64 {
  n * n
}

fn main() {
  let mut total: i64 = 0;
  let mut i: i64 = 0;
  while i < 1_000_000 {
    total += square(i);
    i += 1;
  }
  println!("{total}");
}
